docs/reduction-in-locks.md

```diff
@@ -0,0 +1,100 @@
+
+* Reduced locking: Rather than holding the backup lock throughout the entire backup process, it is only acquired for a very short period.
+
+* DDL statements: The server remains accessible for `DDL` operations during the backup process.
+
+* Consistency: Consistency is maintained by using the redo log to track file operations. During the backup process, new metadata files are created to ensure consistency. In the `--prepare` phase, these metadata files are processed first to bring the database to a consistent state.
+
+The new backup design includes the following phases:
+
+### Phase 1: `Operations performed without the lock`
+
+1. Copy the redo logs from the checkpoint up to the current `LSN` and start tracking new entries.
+2. Start the redo log thread.
+3. Track file operations from the redo log by parsing the `MLOG_FILE_*` records.
+4. Copy the `.ibd` files.
+
+### Phase 2: `Operations performed under the lock`
+
+1. Acquire the backup lock (Lock Tables For Backup (LTFB)/Lock Instance For Backup (LIFB)) to prevent new DDL operations, such as creating or altering tables.
+2. Check the file operations that were tracked and recopy the tablespaces.
+3. Create additional `meta` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This step is required for taking streaming backups.
+    
+    The meta files are the following:
+
+    | File         | Description                                                                                                      |
+    |--------------|------------------------------------------------------------------------------------------------------------------|
+    | `.new`       | For files that need to be recopied due to encryption or `ADD INDEX` operations.                                  |
+    | `.del`       | For deleted files. If a file has been copied, create a `space_id.del` file.                                      |
+    | `.ren`       | For renamed files after copying. Create a `space_id.ren` file.                                                   |
+    | `.crpt`      | For files that cannot be fully copied due to encryption changes. These will be recopied, and a `.crpt` file will be created for the existing file. |
+    | `.new.meta`  | Similar to `.new`, but for incremental backups.                                                                  |
+    | `.new.delta` | Similar to `.new`, but for incremental backups. Create `t1.ibd.meta` and `t1.ibd.delta` instead of `t1.ibd`.      |
+
+4. Gather a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+5. Stop the redo thread once it copies at least up to sync point at step 4.
+6. Release the backup lock (LTFB/LIFB).
+
+### Phase 3. Processing the new metadata files during the `--prepare` phase before crash recovery starts.
```

Comment:
Phase 3 - we commonly say,  "In the `prepare` phase," - so there seem to be too many "phases".  

In Percona XtraBackup, “phase” refers to the distinct stages of the backup process, each handling a specific task to complete a backup or restore. These phases allow XtraBackup to systematically back up, process, and prepare the data for recovery.

Reply:
updated